pkg/providers/vmware/mapper/mapper.go

```diff
@@ -0,0 +1,567 @@
+
+	v1beta1 "github.com/kubevirt/vm-import-operator/pkg/apis/v2v/v1beta1"
+	vos "github.com/kubevirt/vm-import-operator/pkg/providers/vmware/os"
+	"github.com/kubevirt/vm-import-operator/pkg/utils"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/mo"
+	"github.com/vmware/govmomi/vim25/types"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubevirtv1 "kubevirt.io/client-go/api/v1"
+	cdiv1 "kubevirt.io/containerized-data-importer/pkg/apis/core/v1alpha1"
+)
+
+const (
+	cdiAPIVersion                 = "cdi.kubevirt.io/v1alpha1"
+	dataVolumeKind                = "DataVolume"
+	defaultStorageClassTargetName = ""
+	labelTag                      = "tags"
+	vmNamePrefix                  = "vmware-"
+	vmwareDescription             = "vmware-description"
+)
+
+// bus types
+const (
+	busTypeUSB    = "usb"
+	busTypeVirtio = "virtio"
+)
+
+// network types
+const (
+	networkTypeMultus = "multus"
+	networkTypePod    = "pod"
+)
+
+var biosTypeMapping = map[string]*kubevirtv1.Bootloader{
+	"efi":  {EFI: &kubevirtv1.EFI{}},
+	"bios": {BIOS: &kubevirtv1.BIOS{}},
+}
+
+// disk is an abstraction of a VMWare VirtualDisk
+type disk struct {
+	backingFileName string
+	bus             string
+	capacity        resource.Quantity
+	datastore       string
+	id              string
+	name            string
+}
+
+// nic is an abstraction of a VMWare VirtualEthernetCard
+type nic struct {
+	name  string
+	moRef string
+	mac   string
+}
+
+// DataVolumeCredentials defines the credentials required
+// for creating a DataVolume with a VDDK source.
+type DataVolumeCredentials struct {
+	URL        string
+	Username   string
+	Password   string
+	Thumbprint string
+	SecretName string
+}
+
+// VmwareMapper is a struct that holds attributes needed to map a vSphere VM to Kubevirt
+type VmwareMapper struct {
+	credentials    *DataVolumeCredentials
+	disks          *[]disk
+	disksByDVName  map[string]disk
+	hostProperties *mo.HostSystem
+	mappings       *v1beta1.VmwareMappings
+	namespace      string
+	nics           *[]nic
+	osFinder       vos.OSFinder
+	vm             *object.VirtualMachine
+	vmProperties   *mo.VirtualMachine
+}
+
+// NewVmwareMapper creates a new VmwareMapper struct
+func NewVmwareMapper(vm *object.VirtualMachine, vmProperties *mo.VirtualMachine, hostProperties *mo.HostSystem, credentials *DataVolumeCredentials, mappings *v1beta1.VmwareMappings, namespace string, osFinder vos.OSFinder) *VmwareMapper {
+	return &VmwareMapper{
+		credentials:    credentials,
+		disksByDVName:  make(map[string]disk),
+		hostProperties: hostProperties,
+		mappings:       mappings,
+		namespace:      namespace,
+		osFinder:       osFinder,
+		vm:             vm,
+		vmProperties:   vmProperties,
+	}
+}
+
+// buildDevices retrieves each of the VM's VirtualDisks and VirtualEthernetCards
+// and pulls out the values that are needed for import
+func (r *VmwareMapper) buildDevices() error {
```

Comment:
Mostly I was trying to avoid iterating over that array twice, but I think you're right that it would be better to split it up for organizational purposes.